public/docs/_examples/style-guide/ts/001/app/placeholder.ts

```diff
@@ -0,0 +1,93 @@
+// #docplaster
+
+// #docregion 001-1
+  /* avoid */
+  import { bootstrap } from 'angular2/platform/browser';
+  import { Component, OnInit } from 'angular2/core';
+
+  @Component({
+    selector: 'my-app',
+    template: `
+        <h1>{{title}}</h1>
+        <pre>{{heroes | json}}</pre>  
+      `,
+    styleUrls: ['app/app.component.css']
+  })
+  export class AppComponent implements OnInit{
+    title = 'Tour of Heroes';
+
+    heroes: Hero[] = [];
+
+    ngOnInit() {
+      getHeroes().then(heroes => this.heroes = heroes);
+    }
+  }
+
+  bootstrap(AppComponent, []);
+
+  function getHeroes() {
+    return // some promise of data;
+  }  
+// #enddocregion 001-1
+
+
+// #docregion 001-2
+  /* recommended */
+
+  // main.ts
+  import { bootstrap } from 'angular2/platform/browser';
+  import { AppComponent } from './app.component';
+
+  bootstrap(AppComponent, []);
+  /* recommended */
+
+  // app.component.ts
+  import { Component, OnInit } from 'angular2/core';
+
+  import { Hero } from './hero';
+  import { HeroService } from './hero.service';
+
+  @Component({
+    selector: 'my-app',
+    template: `
+        <pre>{{heroes | json}}</pre>  
+      `,
+    styleUrls: ['app/app.component.css'],
+    providers: [HeroService]
+  })
+  export class AppComponent implements OnInit{
+    heroes: Hero[] = [];
+
+    constructor(private _heroService: HeroService) {}
```

Comment:
Just out of curiosity: Why do you still recommend to prefix private fields with `_`. I always saw that as an ugly hack because of the lack of the private modifier, so since using TypeScript I totally dropped that.